graph/patterns/viewpoint.md

```diff
@@ -0,0 +1,112 @@
+# Viewpoint
+
+Microsoft Graph API Design Pattern
+
+
+*The viewpoint pattern provides the ability to manage an individual status of a shared object for multiple independent actors.*
+
+## Problem
+A shared resource, such as a website or a group message, may have different states for different users who access it at different times in an organizational context. For example, user1 may read and delete a message, while user2 may not have seen it yet. This usually happens when a shared item is presented in an individual context.
+## Solution
+
+The viewpoint pattern provides a solution to how to model an individual user context on a shared resource using a `viewpoint` structural property on an API entity type.
+For example, the `viewpoint` property can indicate whether a message is read, deleted, or flagged for a given user. 
+The consistent naming convention ensures that when a developer uses Graph APIs all ‘viewpoints’ structural properties represent type specific user context across different M365 services and features.
+
+This pattern simplifies the API client logic by hiding the state transition details and providing state persistency on the server side. The server can manage the different viewpoints for the shared resource without exposing additional complexity to the client.To support queries for a user state the `viewpoint` property should support be filterable.
+## Issues and considerations
+
+- Because the `viewpoint` property reflects an individual user's context, it is null when accessed with application permissions.
+- Sometimes, the viewpoint can be computed on the server. In this case, an API producer should add OData annotations to the property to provide more information for downstream tools, such as SDKs and documentation generation.
+```
+    <Annotations Target="microsoft.graph.approvalItem/viewPoint">
+        <Annotation Term="Org.OData.Core.V1.Computed" Bool="true" />
+    </Annotations>
+```
+- An alternative to this design would be to store the user state on the client side. However, this may be problematic in some cases, because of the many devices that a user may have and the need to synchronize the state across them.
```

Comment:
I don't understand how storing API state on a client would be a viable solution. What am I missing?

Reply:
like storing personal bookmarks on a client browser instead of managing it for all users on a server

Reply:
I see, so if we removed the state from the API completely.